inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -125,7 +125,7 @@ def __init__(self,
             # If an exception occurs during string conversion, raise that exception
             raise SotaError('package_list is not a string in manifest') from e
 
-        self._device_reboot = parsed_manifest['deviceReboot']
+        self._device_reboot = str(parsed_manifest['deviceReboot'])
```

Comment:
Use a more robust method to convert the `deviceReboot` value to a string that handles potential `None` or unexpected types gracefully. [important]